dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageRenderUtil.java

```diff
@@ -333,6 +354,37 @@ private List<ContainerRaw> populateContainers() throws DotDataException, DotSecu
         return raws;
     }
 
+    private void addRelationships(final Contentlet contentlet) {
+
+        final HttpServletRequest  request  = HttpServletRequestThreadLocal.INSTANCE.getRequest();
+        final HttpServletResponse response = HttpServletResponseThreadLocal.INSTANCE.getResponse();
+        if (ADD_RELATIONSHIPS_ON_PAGE.get() && null != response && null != request && null != request.getParameter("depth")) {
+
+            final int depth = ConversionUtils.toInt(request.getParameter("depth"), -1);
+            if (depth >= 0 && depth <= 3) {
+
+                try {
+
+                    final JSONObject jsonWithRelationShips = ContentResource.addRelationshipsToJSON(request, response,
+                            request.getParameter("render"), user, depth, mode.respectAnonPerms, contentlet,
+                            new JSONObject(), null, languageId, mode.showLive, false,
+                            true);
+
+                    final HashMap<String,Object> relationshipsMap = DotObjectMapperProvider.getInstance()
+                            .getDefaultObjectMapper().readValue(jsonWithRelationShips.toString(), HashMap.class);
+
+                    if (UtilMethods.isSet(relationshipsMap)) {
+                        contentlet.getMap().putAll(relationshipsMap);
+                    }
+                } catch (Exception e) {
+
+                    Logger.error(this, e.getMessage(), e);
```

Comment:
Let's add the Contentlet ID to the error message and/or any other important info